### PR TITLE
McAfeeePO Data Connector

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes
  - Add McAfeeePO Data Connector
  - Add McAfeeePO Workbook

### CODE REVIEW

File: Workbooks/WorkbooksMetadata.json [hunk 1]
```diff
@@ -1270,5 +1270,18 @@
     "templateRelativePath": "WorkspaceUsage.json",
     "subtitle": "",
     "provider": "Azure Sentinel community"
+  },
+  {
+    "workbookKey": "McAfeeePOWorkbook",
+    "logoFileName": "mcafee_logo.svg",
```
Comment (v-jayakal) [inline Workbooks/WorkbooksMetadata.json:1276]: Please add the logo under workbooks/images/logos. You can refer the instructions here https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks (step 2)
(optional) A logo that you want the workbook to display. Place it under workbooks/images/logos - if not supplied - it will be the generic workbooks logo.
This logo should be in SVG format.

File: Workbooks/WorkbooksMetadata.json [hunk 2]
```diff
@@ -1150,7 +1150,7 @@
     "previewImagesFileNames": [ "IntsightsIOCWhite.png", "IntsightsMatchedWhite.png", "IntsightsMatchedBlack.png", "IntsightsIOCBlack.png"],
     "version": "2.0",
     "title": "IntSights IOC Workbook",
-    "templateRelativePath": "IntsightsIOCWorkbook.json",
```
Comment (v-jayakal) [inline Workbooks/WorkbooksMetadata.json:1153]: Can you please undo this change, even though it is simple space removal.

File: DataConnectors/McAfeeePO/Connector_McAfee_ePO.json [hunk 3]
```diff
@@ -0,0 +1,124 @@
+{
+    "id": "McAfeeePO",
+    "title": "McAfee ePolicy Orchestrator (ePO)",
+    "publisher": "McAfee",
+    "descriptionMarkdown": "The McAfee ePolicy Orchestrator data connector provides the capability to ingest [McAfee ePO](https://www.mcafee.com/enterprise/en-us/products/epolicy-orchestrator.html) events into Azure Sentinel through the syslog. Refer to [documentation](https://docs.mcafee.com/bundle/epolicy-orchestrator-landing/page/GUID-0C40020F-5B7F-4549-B9CC-0E017BC8797F.html) for more information.",
+    "additionalRequirementBanner": "This data connector depends on a parser based on Kusto Function to work as expected. Follow the steps to use this Kusto Function alias **McAfeeEPOEvent** in queries and workbooks. [Follow steps to get this Kusto Function>](https://aka.ms/sentinel-McAfeeePO-parser)\nNote: This data connector has been developed using McAfee ePolicy Orchestrator 5.10.x (Syslog)",
```
Comment (v-jayakal) [inline DataConnectors/McAfeeePO/Connector_McAfee_ePO.json:6]: Could not find the parser McAfeeEPOEvent in Sentinel Github (if already merged) or a PR, Can you please provide link to the Parser mentioned? This is needed for all queries validation.

File: DataConnectors/McAfeeePO/Connector_McAfee_ePO.json [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    ],
+    "sampleQueries": [
+        {
+            "description" : "Top 10 Sources",
+            "query": "McAfeeEPOEvent\n | summarize count() by DvcHostname\n | top 10 by count_"
+        }
+    ],
+    "dataTypes": [
+        {
+            "name": "Syslog(McAfee ePO)",
```
Comment (v-jayakal) [inline DataConnectors/McAfeeePO/Connector_McAfee_ePO.json:22]: DATATYPE_NAME does not have spaces, Can you please remove space in "McAfee ePO" and make it "McAfeeePO".

File: DataConnectors/McAfeeePO/Connector_McAfee_ePO.json [hunk 5]
```diff
@@ -0,0 +1,124 @@
+{
+    "id": "McAfeeePO",
+    "title": "McAfee ePolicy Orchestrator (ePO)",
+    "publisher": "McAfee",
+    "descriptionMarkdown": "The McAfee ePolicy Orchestrator data connector provides the capability to ingest [McAfee ePO](https://www.mcafee.com/enterprise/en-us/products/epolicy-orchestrator.html) events into Azure Sentinel through the syslog. Refer to [documentation](https://docs.mcafee.com/bundle/epolicy-orchestrator-landing/page/GUID-0C40020F-5B7F-4549-B9CC-0E017BC8797F.html) for more information.",
+    "additionalRequirementBanner": "This data connector depends on a parser based on Kusto Function to work as expected. Follow the steps to use this Kusto Function alias **McAfeeEPOEvent** in queries and workbooks. [Follow steps to get this Kusto Function>](https://aka.ms/sentinel-McAfeeePO-parser)\nNote: This data connector has been developed using McAfee ePolicy Orchestrator 5.10.x (Syslog)",
+    "graphQueries": [
+        {
+            "metricName": "Total data received",
+            "legend": "McAfee ePO Events",
```
Comment (v-jayakal) [inline DataConnectors/McAfeeePO/Connector_McAfee_ePO.json:10]: Should be DATATYPE_NAME, which is "McAfeeePO".

**Review by v-jayakal — CHANGES_REQUESTED** [review]

**Comment by v-jayakal** [comment]
@vu-socprime - Can you please place a distinct name related to McAfeeePO for 'fromTemplateId' attrinbute in McAfeeePOOverview.json? It should not be generic value of "sentinel-UserWorkbook". You can check other workbooks for reference.

File: DataConnectors/McAfeeePO/Connector_McAfee_ePO.json [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+                "providerDisplayName": "Workspace",
+                "scope": "Workspace",
+                "requiredPermissions": {
+                    "write": true,
+                    "read": true,
+                    "delete": true
+                }
+            },
+            {
+                "provider": "Microsoft.OperationalInsights/workspaces/sharedKeys",
```
Comment (v-admahe) [inline DataConnectors/McAfeeePO/Connector_McAfee_ePO.json:52]: Please remove shared Keys section from the permission. Please refer the sample template : https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_Syslog_template.json

or you can use this :
 "permissions": {
        "resourceProvider": [
            {
                "provider": "Microsoft.OperationalInsights/workspaces",
                "permissionsDisplayText": "write permission is required.",
                "providerDisplayName": "Workspace",
                "scope": "Workspace",
                "requiredPermissions": {
                    "write": true,
                    "delete": true
                }
            }
        ]
    },

Thanks,
Aditi Maheshwari

**Review by v-admahe — DISMISSED** [review]

**Comment by vu-socprime** [comment]
> @vu-socprime - Can you please place a distinct name related to McAfeeePO for 'fromTemplateId' attrinbute in McAfeeePOOverview.json? It should not be generic value of "sentinel-UserWorkbook". You can check other workbooks for reference.

done

**Review by v-jayakal — CHANGES_REQUESTED** [review]

**Review by v-jayakal — APPROVED** [review]

**Comment by v-jayakal** [comment]
@vu-socprime - Can you please resolve the merge conflicts?

File: Workbooks/WorkbooksMetadata.json [hunk 7]
```diff
@@ -1270,5 +1270,18 @@
     "templateRelativePath": "WorkspaceUsage.json",
     "subtitle": "",
     "provider": "Azure Sentinel community"
+  },
+  {
+    "workbookKey": "McAfeeePOWorkbook",
```
Comment (v-jayakal) [inline Workbooks/WorkbooksMetadata.json:1275]: There are merge conflicts, please resolve them.

**Review by v-jayakal — CHANGES_REQUESTED** [review]

**Review by v-jayakal — APPROVED** [review]
